ee/debug/checkups/runtime.go

```diff
@@ -133,3 +145,232 @@ func gatherPprofCpu(z *zip.Writer) error {
+}
+
+type desktopProcessInfo struct {
+	socketPath string
+	authToken  string
+}
+
+func (c *runtimeCheckup) testDesktopProfilingSupport(socketPath, authToken string) bool {
+	// Quick test to see if this desktop process supports profiling endpoints
+	// We'll try a simple ping first to avoid auth errors on old processes
+	client := client.New(authToken, socketPath, client.WithTimeout(5*time.Second))
+
+	// Test with a simple ping first
+	if err := client.Ping(); err != nil {
+		return false // Can't even ping, skip this process
+	}
+
+	// Now test if profiling endpoints exist by making a quick HTTP request
+	// We'll check if the endpoint exists without actually generating a profile
+	testClient := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, network, addr string) (net.Conn, error) {
+				return net.Dial("unix", socketPath)
+			},
+		},
+		Timeout: 5 * time.Second,
+	}
+
+	// Test cpuprofile endpoint with OPTIONS to see if it exists
+	req, err := http.NewRequestWithContext(context.Background(), "OPTIONS", "http://unix/cpuprofile", nil)
+	if err != nil {
+		return false
+	}
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", authToken))
+
+	resp, err := testClient.Do(req)
+	if err != nil {
+		return false
+	}
+	defer resp.Body.Close()
+
+	// If we get anything other than 404, the profiling endpoints likely exist
+	// 404 = endpoint doesn't exist, others suggest profiling support exists
+	return resp.StatusCode != http.StatusNotFound
+}
+
+func gatherDesktopProfilesFromSocket(ctx context.Context, z *zip.Writer, socketPath, authToken string, processIndex int) error {
+	// Create a client to communicate with the desktop server
+	c := client.New(authToken, socketPath, client.WithTimeout(30*time.Second))
+
+	// Try to ping first to see if the server is responsive
+	if err := c.Ping(); err != nil {
+		return fmt.Errorf("desktop server not responsive: %w", err)
+	}
+
+	// Request CPU profile
+	cpuProfilePath, err := requestDesktopProfile(socketPath, authToken, "cpuprofile")
+	if err != nil {
+		return fmt.Errorf("requesting CPU profile: %w", err)
+	}
+	defer os.Remove(cpuProfilePath) // Clean up temp file
+
+	// Request memory profile
+	memProfilePath, err := requestDesktopProfile(socketPath, authToken, "memprofile")
+	if err != nil {
+		return fmt.Errorf("requesting memory profile: %w", err)
+	}
+	defer os.Remove(memProfilePath) // Clean up temp file
+
+	// Add CPU profile to zip
+	if err := addFileToZipFromPath(z, cpuProfilePath, fmt.Sprintf("desktop_%d_cpuprofile", processIndex)); err != nil {
+		return fmt.Errorf("adding desktop CPU profile to zip: %w", err)
+	}
+
+	// Add memory profile to zip
+	if err := addFileToZipFromPath(z, memProfilePath, fmt.Sprintf("desktop_%d_memprofile", processIndex)); err != nil {
+		return fmt.Errorf("adding desktop memory profile to zip: %w", err)
+	}
+
+	return nil
+}
+
+func requestDesktopProfile(socketPath, authToken, profileType string) (string, error) {
```

Comment:
Let's move this into ee/desktop/user/client/client.go with the other desktop process request functions?